standard-controller/src/main/java/io/enmasse/controller/standard/AddressProvisioner.java

```diff
@@ -90,7 +90,30 @@ private void addToUsage(Map<String, Map<String, UsageInfo>> usageMap, Address ad
         return newUsageMap;
     }
 
-    private Map<String, Map<String, UsageInfo>> checkQuotaForAddress(Map<String, Double> limits, Map<String, Map<String, UsageInfo>> usage, Address address) {
+    private Address findAddress(String name, Set<Address> addressSet) {
+        for (Address address : addressSet) {
+            if (name.equals(address.getAddress())) {
+                return address;
+            }
+        }
+        return null;
+    }
+
+    private void checkBrokerQuotaForSubscription(Address subscription, Address topic, Map<String, UsageInfo> brokerUsage, ResourceRequest requested) {
+        String cluster = topic.getAnnotations().get(AnnotationKeys.CLUSTER_ID);
+        String broker = topic.getAnnotations().get(AnnotationKeys.BROKER_ID);
+        subscription.getAnnotations().put(AnnotationKeys.CLUSTER_ID, cluster);
+        if (broker != null) {
+            brokerUsage.get(broker).addUsed(requested.getAmount());
+            subscription.getAnnotations().put(AnnotationKeys.BROKER_ID, broker);
+        } else {
+            brokerUsage.get(cluster).addUsed(requested.getAmount());
+            log.info("need to allocate subscription {} on {} to broker from {}", subscription.getAddress(), topic.getAddress(), cluster);
+            //TODO: find least used broker on topic cluster
```

Comment:
I think in this place, we should do something similar to 'scheduleAddress' which sorts a list of brokers by usage and allocates to the least used broker.

However, I guess the problem here is that the UsageInfo for sharded address plans only tracks the usage of the cluster in sum, and not per broker (see https://github.com/EnMasseProject/enmasse/blob/ce82d1bcaa042e8718c5706b533c9f250f9847ab/standard-controller/src/main/java/io/enmasse/controller/standard/AddressProvisioner.java#L50). 

The usage of a sharded broker would be the `creditsUsedByAddress + sumCreditsForSubscriptionsOnBroker`.